lib/scripting/objects/pinball-backend/rom-fetcher.ts

```diff
@@ -0,0 +1,122 @@
+import { GamelistDB } from 'wpc-emu';
+import { logger } from '../../../util/logger';
+
+/**
+ * Functions to fetch a WPC ROM file from VPDB.io
+ */
+
+export function downloadGameEntry(pinmameGameName: string): Promise<LoadedGameEntry> {
+	const gameEntry: undefined | GamelistDB.ClientGameEntry = GamelistDB.getByPinmameName(pinmameGameName);
+	if (!gameEntry) {
+		return Promise.reject(new Error('GAME_ENTRY_NOT_FOUND_' + pinmameGameName));
+	}
+	const url: string = buildVpdbGameEntryUrl(gameEntry.pinmame.vpdbId || gameEntry.pinmame.id);
+	return downloadFileAsJson(url)
+		.then((jsonData: VpdbGameEntry[]) => {
+			if (!Array.isArray(jsonData)) {
+				return Promise.reject(new Error('VPDB_INVALID_ANSWER'));
+			}
+			const result: VpdbGameEntry | void = jsonData.find((vpdbEntry: VpdbGameEntry) => vpdbEntry.id === pinmameGameName);
+			if (!result) {
+				return Promise.reject(new Error('VPDB_GAME_ENTRY_NOT_FOUND'));
```

Comment:
`throw`

Maybe add which ROM it was looking for and the game name so we can more easily debug it?

Reply:
like that!